sql/src/lib.rs

```diff
@@ -69,11 +113,70 @@ impl QueryBuilder for SQLiteQueryBuilder {
         Ok(())
     }
 
-    fn push_bind_param(&mut self) {
-        self.push_sql("?");
+    fn push_typed_value(&mut self, value: &TypedValue) -> BuildQueryResult {
+        use TypedValue::*;
+        match value {
```

Comment:
Opinions in `#rust` were divided. ("A good question for StackOverflow")

It's possible ("check the asm") that the `match *v` version could copy, or could move, depending on the match arms. The `match v` version should never be bad. If the ASM disagrees, then it becomes personal preference.

Opinion was also somewhat against Clippy: as with all linters, it doesn't always give the best advice. Quoth:

> Manishearth: clippy isn't intended to be blindly followed

So, given that I'm too lazy to disassemble Mentat to find out if using `match *foo` produces worse assembly, I'll stick with `match foo`.